packages/browser-wallet/src/popup/shared/ContractTokenLine/ContractTokenLine.tsx

```diff
@@ -0,0 +1,74 @@
+import clsx from 'clsx';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { ClassNameAndStyle } from 'wallet-common-helpers';
+import Button from '@popup/shared/Button';
+import { Checkbox } from '@popup/shared/Form/Checkbox';
+import TokenBalance from '@popup/shared/TokenBalance';
+import { ContractTokenDetails } from '@shared/utils/token-helpers';
+
+export enum ChoiceStatus {
+    discarded,
+    chosen,
+    existing,
+}
+
+type ContractTokenLineProps = ClassNameAndStyle & {
+    token: ContractTokenDetails;
+    onClick(token: ContractTokenDetails): void;
+    onToggleChecked(token: ContractTokenDetails): void;
+    status: ChoiceStatus;
+};
+
+export default function ContractTokenLine({
+    token,
+    onClick,
+    onToggleChecked: toggleChecked,
+    status,
+    className,
+    style,
+}: ContractTokenLineProps) {
+    const { t } = useTranslation('account', { keyPrefix: 'tokens.add' });
+    const clickable = status !== ChoiceStatus.existing;
+
+    return (
+        <Button
+            key={token.id}
+            clear
+            className={clsx('contract-token-line__token', className)}
+            style={style}
+            onClick={() => onClick(token)}
+        >
+            <div className="flex align-center h-full">
+                <img src={token.metadata.thumbnail?.url} alt={token.metadata.name ?? ''} />
+                <div>
+                    {token.metadata.name}
+                    <div
+                        className={clsx(
+                            'contract-token-line__token-balance',
+                            token.balance !== 0n && 'contract-token-line__token-balance--owns'
+                        )}
+                    >
+                        {t('ItemBalancePre')}
+                        <TokenBalance
+                            balance={token.balance}
+                            decimals={token.metadata.decimals ?? 0}
+                            symbol={token.metadata.symbol}
+                        />
+                    </div>
+                </div>
+            </div>
+            {clickable && (
+                <Checkbox
+                    onClick={(e) => {
+                        e.stopPropagation();
+                    }}
+                    onChange={() => toggleChecked(token)}
+                    checked={status === ChoiceStatus.chosen}
+                    className="contract-token-line__checkbox"
+                />
+            )}
+            {clickable || <p className="m-l-auto text-faded">Already Added</p>}
```

Comment:
I think the following looks alot better. If you don't agree, we at least need a translation here for "Already added".

Linter will probably reject if you just apply the suggestion (spacing)

```suggestion
                <Checkbox
                    onClick={(e) => {
                        e.stopPropagation();
                    }}
                    onChange={() => toggleChecked(token)}
                    checked={status === ChoiceStatus.chosen}
                    className="contract-token-line__checkbox"
                    disabled={clickable}
                />
```

Reply:
I think that might be slightly better. I've added it (though made the box checked when already added)